record/record_test.go

```diff
@@ -312,3 +313,146 @@ func TestStaleWriter(t *testing.T) {
 		t.Fatalf("stale write #1: unexpected error: %v", err)
 	}
 }
+
+func TestBasicRecover(t *testing.T) {
+	records := [][]byte{
+		[]byte(strings.Repeat("a", blockSize-headerSize)),
+		[]byte(strings.Repeat("b", blockSize-headerSize)),
+		[]byte(strings.Repeat("c", blockSize-headerSize)),
+	}
+
+	buf := new(bytes.Buffer)
+	w := NewWriter(buf)
+
+	for i := 0; i < len(records); i++ {
+		wRec, err := w.Next()
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err = wRec.Write(records[i]); err != nil {
+			t.Fatal(err)
+		}
+	}
+	w.Close() // Finalize the last record.
+
+	rawBufSlice := buf.Bytes()
```

Comment:
I'd roll this into the next block of code:

----
w.Close()

// Corrupt the checksum of the second record in our file.
rawBufSlice := buf.Bytes()
rawBufSlice[blockSize+0] = 0xef
rawBufSlice[blockSize+1] = 0xbe
etc
----

Similarly below.